File: 2309-greatest-english-letter-in-upper-and-lower-case.py
class Solution:
    def greatestLetter(self, s: str) -> str:
        flag1, flag2 = [False] * 26, [False] * 26
        for ch in s:
            if ch.isupper():
                flag1[ord(ch) - ord('A')] = True
            else:
                flag2[ord(ch) - ord('a')] = True
        for i in range(25, -1, -1):
            if flag1[i] and flag2[i]:
                return chr(ord('A') + i)
        return ""


solution = Solution()
print(solution.greatestLetter("lEeTcOdE"))
print(solution.greatestLetter("arRAzFif"))
print(solution.greatestLetter("AbCdEfGhIjK"))

File: 2309-greatest-english-letter-in-upper-and-lower-case.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
	string greatestLetter(string s) {
		int flag[26] = { 0 };
		for (auto &ch : s) {
			if (isupper(ch)) {
				flag[ch - 'A'] |= 2;
			}
			else {
				flag[ch - 'a'] |= 1;
			}
		}
		for (int i = 25; i >= 0; --i){
			if (flag[i] == 3) {
				return string(1, ('A' + i));
			}
		}
		return "";
	}
};

int main(void) {
	Solution solution = Solution();
	cout << solution.greatestLetter("lEeTcOdE") << endl;
	cout << solution.greatestLetter("arRAzFif") << endl;
	cout << solution.greatestLetter("AbCdEfGhIjK") << endl;
}